engine/src/main/java/org/hibernate/search/spi/SearchIntegratorBuilder.java

```diff
@@ -403,6 +407,35 @@ else if ( metadataProvider.containsSearchMetadata( mappedClass ) ) {
 	}
 
 	/**
+	 * Get the filter definitions from the super classes that are not annotated with @Indexed
+	 */
+	private void bindFilterDefsInSuperClasses(final Map<XClass, Class<?>> classMappings) {
+		Set<XClass> superClasses = new HashSet<XClass>();
+		for ( Map.Entry<XClass, Class<?>> mapping : classMappings.entrySet() ) {
+			saveSuperClasses( superClasses, mapping.getKey() );
+		}
+		for ( XClass xClass : superClasses ) {
+			bindFilterDefs( xClass );
+		}
+	}
+
+	/**
+	 * It will add in the list all the super classes of a given class to the collection
+	 * if it's not annotated with @Indexed and it's not already in the collection.
+	 *
+	 * @param superClasses the list containing the super classes
+	 * @param mappedXClass the current class
+	 */
+	private void saveSuperClasses(Set<XClass> superClasses, XClass mappedXClass) {
+		if ( mappedXClass.getSuperclass() != null ) {
+			saveSuperClasses( superClasses, mappedXClass.getSuperclass() );
+		}
+		if ( !mappedXClass.isAnnotationPresent( Indexed.class ) && !OBJECT_CLASS_NAME.equals( mappedXClass.getName() ) ) {
```

Comment:
why the condition of `!mappedXClass.isAnnotationPresent( Indexed.class )`? Why can it not be annotated with `@Indexed`. Filter definitions are global and the the only way to identify them is by their unique name. It does not matter where the definition is placed. For simplicity we always assumed that we just check the configured class itself, but now we want to consider the whole hierarchy. 

That said, there is a `ReflectionHelper.createXClassHierarchy( parseContext.getCurrentClass() );` which gives you the hierarchy as a list (bottom up). I think you could use that.

Last but not least, we are already processing the full hierarchy for indexed classes in `AnnotationMetadataProvider`. Now we do it twice as bootstrap. Once for the filter definitions and once when we process the rest of the configuration. Seems to be a waste. Technically there would be no problem to find the filter definitions in `AnnotationMetadataProvider` instead and only process the class hierarchies once.